ff_control/scripts/safety_filter

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python3
+
+# MIT License
+#
+# Copyright (c) 2023 Stanford Autonomous Systems Lab
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+
+import rclpy
+from rclpy.node import Node
+from rclpy.duration import Duration
+import numpy as np
+import time
+
+from ff_msgs.msg import ThrusterCommand
+from ff_msgs.msg import WheelVelCommand
+from std_msgs.msg import Bool
+
+class SafetyFilter(Node):
+
+    def __init__(self):
+        super().__init__('safety_filter')
+        self.get_logger().info("Safety Filter Node Started")
+        self.last_thrust_time = None
+        self.last_wheel_time = None
+        self.wait_period = 1 # seconds to wait w/ receiving anything before publishing a zero message
+        self.kill_state = False
+
+        # Create publishers for thruster and wheel commands
+        self.thrust_pub = self.create_publisher(ThrusterCommand, 'commands/duty_cycle', 10)
+        self.wheel_pub = self.create_publisher(WheelVelCommand, 'commands/velocity', 10)
+
+        # Create subscribers for thruster and wheel commands
+        self.thrust_sub = self.create_subscription(
+            ThrusterCommand,
+            'ctrl/duty_cycle',
+            self.thrust_callback,
+            10)
+        self.wheel_sub = self.create_subscription(
+            WheelVelCommand,
+            'ctrl/velocity',
+            self.wheel_callback,
+            10)
+
+        # Create timer to check if we've received any messages in the last 'wait_period' seconds
+        self.timer = self.create_timer(0.1, self.check_timer_callback)
+
+        # Create a subscriber to listen for a kill message
+        self.kill_sub = self.create_subscription(
+            Bool,
+            'kill',
+            self.kill_callback,
+            10)
```

Comment:
Is there a publisher for the 'kill' channel? I didn't see one in the codebase and don't know of a universal trigger for publishing to 'kill'. Or is the idea to manually publish from the terminal?